tests/credential_plugins_test.py

```diff
@@ -134,6 +135,67 @@ def test_hashivault_handle_auth_not_enough_args() -> None:
         hashivault.handle_auth()
 
 
+@pytest.mark.parametrize(
+    'kwargs',
+    (
+        {
+            'access_key': 'my_access_key',
+            'secret_key': 'my_secret_key',
+            'role_arn': 'the_arn',
+            'identifier': 'access_token',
+        },
+    ),
+    (
+        {
+            'role_arn': 'the_arn',
+            'identifier': 'access_token',
+        },
+    ),
+)
+@pytest.mark.parametrize(
+    (
+        'identifier_key',
+        'expected',
+    ),
+    (
+        (None, 'the_access_token'),
+        ('access_key', 'the_access_key'),
+        ('secret_key', 'the_secret_key'),
+    ),
+)
+def test_aws_assumerole_identifier(
+    monkeypatch: pytest.MonkeyPatch,
+    kwargs: dict, identifier_key: str, expected: dict[str, str],
```

Comment:
Looking into the decorators above, the actual types are
```suggestion
    kwargs: dict[str, str], identifier_key: str | None, expected: str,
```